docs/releases/upgrading/upgrade-v12.0.md

```diff
@@ -0,0 +1,66 @@
+# Upgrading from v11 to v12
+
+## Prerequisites
+
+The steps listed in this article require an existing local installation of InvenioRDM v11, please make sure that this is given!
+
+If unsure, run `invenio-cli install` from inside the instance directory before executing the listed steps.
+
+!!! warning "Backup"
+
+    Always backup your database and files before you try to perform an upgrade.
+
+!!! info "Older versions"
+
+    In case you have an InvenioRDM installation older than v11, you can gradually upgrade using the existing instructions to v11 and afterwards continue from here.
+
+## Upgrade Steps
+
+!!! warning "Upgrade your invenio-cli"
+
+    Make sure you have the latest `invenio-cli`, for InvenioRDM v12 the release is v1.0.20.
+
+    ```bash
+    $ invenio-cli --version
+    invenio-cli, version 1.0.20
+    ```
+
+!!! info "Virtual environments"
+    In case you are not inside a virtual environment, make sure that you prefix each `invenio` command with `pipenv run`.
+
+### Installing the Latest Versions
+
+Bump the RDM version and rebuild the assets:
+
+```bash
+invenio-cli packages update 12.0.0
+```
+
+These commands should take care of locking the dependencies for v12, installing them, and building the required assets.
+
+### Database chema upgrade
+
+Run the database schema upgrade:
+
+```bash
+# Perform the database upgrade
+pipenv run invenio alembic upgrade
+```
+
+### Initialize new queues
+
+Initialize the new queues with:
+
+```bash
+# Perform the initialization of the new queues
+pipenv run invenio queues declare
+```
+
+### Update OS index
+
+Communities schema was expanded with a new attribute and in consequence the mapping needs to be updated accordingly on the fly.
+
+```bash
+# Perform the update
+curl -X PUT -k -u <your_user>:<your_password> "https://<your_search_host>/es/<your_community_index>/_mapping?pretty" -H 'Content-Type: application/json' -d'{"properties": {"access": {"properties": {"review_policy": {"type": "keyword"}}}}}'
```

Comment:
the modules were released, but the main damage was still not done, so IMO we could add the versions, even if a bit too late. 
I agree, the correct way would be b). We need to pay attention for this in the reviews and releases in the future.
Ping @ntarocco for opinion too

Reply:
Since these are just additions (i.e. they're backwards compatible), wouldn't the normal `invenio index init` just add the fields to the mapping? I think is how we've been doing index migrations in the past for Zenodo.

Reply:
Wasn't this discussed among us when I was adding the `review_policy` field and we aligned that we keep the version? You don't have to reindex all your records but just make sure the mapping is a keyword instead of text....unless I miss some context....

Reply:
Regarding the `invenio index init`, it fails if the index exists. It does not update it. 

```shell
(zenodo-rdm) ~/Workspace/zenodo/zenodo-rdm (sorting ✗) invenio index init
[2023-03-22 11:25:24,611] WARNING in ext: JSONSCHEMAS_HOST is set to localhost
[2023-03-22 11:25:25,409] WARNING in ext: JSONSCHEMAS_HOST is set to localhost
Creating indexes...
  [------------------------------------]    0%
Traceback (most recent call last):

    ensure_not_exists(alias_result[0])
  File "/Users/ppanero/.virtualenvs/zenodo-rdm/lib/python3.9/site-packages/invenio_search/ext.py", line 346, in ensure_not_exists
    raise IndexAlreadyExistsError(
invenio_search.errors.IndexAlreadyExistsError: index/alias with name "zenodo-communities-communities-v1.0.0" already exists
```

(More discussions on Discord)

Extra: purely on versioning, if the change is back compat we should bump minor (same as we do for repos). Otherwise, should we just have just v1, v2, etc. not three digit versioning

Reply:
@ntarocco 

> hidden assumption there that the cluster is protected via basic auth, which might not be the case.

If we do programmatically, like for DB upgrades we can run it within an invenio shell and get the search client connection (i.e. however it is configured by the systadmin)

Reply:
Is this something that we have to do then?

Reply:
Decide that @kpsherva will have a look if it is hard to add a new command in `invenio-indexer` to update mappings.